### PR TITLE
bumps Typescript and its target version

### PR DESCRIPTION
**Pull request by andresgnlez**
Bumps Typescript's target to `ES2020`. The previous target `ES6` was really old and modern browsers already support most of the features included in `ES2020`.

Also, bumps `typescript` to `v5.0`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -15,16 +15,16 @@ module.exports = {
...
-    '@typescript-eslint/no-unsafe-member-access': 'warn',
-    '@typescript-eslint/no-unsafe-call': 'warn',
-    '@typescript-eslint/no-unsafe-argument': 'warn',
-    '@typescript-eslint/no-floating-promises': 'warn',
-    '@typescript-eslint/restrict-template-expressions': 'warn',
-    '@typescript-eslint/no-unsafe-return': 'warn',
-    '@typescript-eslint/ban-ts-comment': 'warn',
-    '@typescript-eslint/no-misused-promises': 'warn',
-    '@typescript-eslint/require-await': 'warn',
+    // '@typescript-eslint/no-unsafe-assignment': 'warn',
```
Comment (Bluesmile82) [inline .eslintrc.js:18]: If we don't need these anymore I think we should remove the comments

File: .eslintrc.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
+/** @type import('eslint').Linter.Config */
 module.exports = {
   extends: [
-    'next/core-web-vitals',
     'plugin:@typescript-eslint/recommended',
-    'plugin:@typescript-eslint/recommended-requiring-type-checking',
```
Comment (andresgnlez) [inline .eslintrc.js:5]: Why are we removing this plugin?

File: .eslintrc.js (hunk 1 above)
Comment (davidsingal) [reply .eslintrc.js:18]: I don't know if we need it. I want to remove it but I don't know if someone thinks this is interesting info.

File: .eslintrc.js (hunk 2 above)
Comment (davidsingal) [reply .eslintrc.js:5]: I thought the `recommended` plugin is already integrating those rules. Let me check again, if not I'll recover it

**Comment by andresgnlez** [comment]
There are some issues to investigate when we are bumping `typescript` and its dependencies (`@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser`) to latest version. Let's be sure those issues are fixed before merging.

**Review by davidsingal — APPROVED** [review]

**Pull request merged**